Yosemite/Yosemite/Stores/BlazeStore.swift

```diff
@@ -120,3 +122,61 @@ private extension BlazeStore {
         }
     }
 }
+
+// MARK: - Synchronize target devices
+private extension BlazeStore {
+
+    func synchronizeTargetDevices(siteID: Int64, onCompletion: @escaping (Result<[BlazeTargetDevice], Error>) -> Void) {
+        Task { @MainActor in
+            do {
+                let stubbedResult = [
+                    BlazeTargetDevice(id: "mobile", name: "Mobile"),
+                    BlazeTargetDevice(id: "desktop", name: "Desktop")
+                ]
+                // TODO-11540: remove stubbed result when the API is ready.
+                let devices: [BlazeTargetDevice] = try await mockResponse(stubbedResult: stubbedResult, onExecution: {
+                    try await remote.fetchTargetDevices(for: siteID)
+                })
+                upsertStoredTargetDeviceInBackground(readonlyDevices: devices) {
+                    onCompletion(.success(devices))
+                }
+            } catch {
+                onCompletion(.failure(error))
+            }
+        }
+    }
+
+    /// Updates or Inserts specified BlazeTargetDevice Entities in a background thread
```

Comment:
"Updates or Inserts" - Should we remove the word "Updates" as we are always deleting any existing items?

Reply:
Also updated in af8228be109a34fa700db2b6208ce486c219e189.